internal/service/create/create.go

```diff
@@ -225,18 +225,22 @@ func NewService(moduleConfigService ModuleConfigService,
 	}, nil
 }
 
-func (s *Service) Run(opts Options) error {
+func (s *Service) Run(opts Options) (rErr error) { //nolint:nonamedreturns // named return to detect error in defer
 	if err := opts.Validate(); err != nil {
 		return err
 	}
 
-	defer s.cleanupTempFiles(opts)
-
 	moduleConfig, err := s.moduleConfigService.ParseAndValidateModuleConfig(opts.ConfigFile)
 	if err != nil {
 		return fmt.Errorf("failed to parse module config: %w", err)
 	}
 
+	defer func() {
+		if rErr != nil { // only clean up if an error occurs
+			s.cleanupTempFiles(opts)
+		}
+	}()
+
```

Comment:
Note to reviewer: As far as I know the named return is the only reliable way to "capture" the returned error value.
I am focusing here only on the "component-constructor mode", as the currently supported "registry push mode" is going to be deprecated in the near future.
That's why `cleanupTempFiles` is invoked only if the function returns an error - it is exactly the logic that is valid for "component-constructor mode".
Essentially I am trying to simplify the removal of "registry push mode" by writing  new code as if "registry push mode"  doesn't exist - then we don't have to adjust this code in the "removal' PR.
Feel free to suggest a nicer way to accomplish that.